pallets/collective-proxy/src/lib.rs

```diff
@@ -85,6 +121,25 @@ pub mod pallet {
         CollectiveProxyExecuted { result: DispatchResult },
     }
 
+    #[pallet::error]
+    pub enum Error<T> {
+        /// There are too many proxies registered
+        TooManyProxies,
+        /// Proxy registration not found.
+        NotFound,
+    }
+
+    /// The set of account proxies
+    #[pallet::storage]
+    pub type Proxies<T: Config> = StorageValue<
```

Comment:
Would you consider using a `StorageMap`, as Dino suggested? This can enforce uniqueness and efficient lookups. Also, it could simplify logic ensuring no redundant filters are stored.

Reply:
Yes, reworked